### PR TITLE
[MMI] Added note to trader code fencing

### PR DESCRIPTION
**Pull request by albertolive**
## Explanation

This PR adds the MMI note-to-trader logic, with code fencing.

https://consensyssoftware.atlassian.net/browse/MMI-2583

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue

### CODE REVIEW

**Comment by zone-live** [comment]
Placing this PR to review after fixing conflicts and adding necessary logic. 👍🏼

**Review by zone-live — DISMISSED** [review]

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js
Comment (brad-decker) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:527]: @jpuri could you review this and see how it fits into your plans for confirmation team/refactors?

**Review by brad-decker — CHANGES_REQUESTED** [review]
Requesting changes to block for Jyoti's review. the code looks good to me overall

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js
Comment (jpuri) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:527]: Thanks a lot @brad-decker , this class ConfirmTransactionBase has become very large due to code being added to it like above. We are refactoring this to remove code like `renderNote` into separate components. I would suggest moving this component outside `ConfirmTransactionBase`.

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js
Comment (jpuri) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:657]: The code here and also in container component can be moved to component `ConfirmHexData`. It is functional component and selectors can be used to get data from state.

**Review by jpuri — CHANGES_REQUESTED** [review]
Thanks @brad-decker : 

I suggested some changes from perspective to keep confirmations code clean and maintainable.

Also we always add unit test cases for confirmations code.

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js
Comment (zone-live) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:657]: @jpuri Not sure I'm following, because in there we want to add a metadata obj to the txData, in order to be send a few [lines bellow here](https://github.com/MetaMask/metamask-extension/blob/4eaee49d1ceee8ff4d7d5a00d9fe97c88b192a26/ui/pages/confirm-transaction-base/confirm-transaction-base.component.js#L711) at `sendTransaction(txData)`

**Comment by zone-live** [comment]
@SocketSecurity ignore-all

File: ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js
Comment (ritave) [inline ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js:5]: Those type of fencing will be a problem if someone wants to add a new build type and be confused why it's not building normally.

It's not a stopper and fine to merge as is, but if you have the time, you could consider if a better solution exists.

* One would be to have `ONLY_EXCLUDE_IN` directive in [remove-fenced-code.js](https://github.com/MetaMask/metamask-extension/blob/37b481f3056344de6d9c763ebeeaccc18146642e/development/build/transforms/remove-fenced-code.js#L124)
* Another one would be have an empty feature such as `group-not-mmi`
* Or just include Button in MMI

Comment (albertolive) [reply ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js:5]: @brad-decker @zone-live wdyt?

File: package.json
Comment (jpuri) [inline package.json:300]: Where is this new dependency used `browserify-zlib`

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js
Comment (jpuri) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:632]: Hey, sorry I has confusion that this was part of renderHexData. I agree that is way to do it.

**Comment by jpuri** [comment]
@zone-live : the latest changes on the PR look good 👍 

The only thing I still see lacking is test coverage, this PR is adding lot of new code with no new tests. Confirmation code is critical and we try to have 80% coverage in all new PRs that we create.

File: package.json
Comment (zone-live) [reply package.json]: A bunch of changes have been reverted already, I had previously merged another PR by mistake that brought unneeded changes. 🤦🏼

File: ui/pages/confirm-transaction-base/confirm-transaction-base.container.js
Comment (albertolive) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.container.js]: Remove console log @zone-live

**Comment by zone-live** [comment]
Hey @jpuri simplified the PR and added an extra test, can I get your approval ✅ ?

**Review by jpuri — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**
